From 436dfddb4c2932ba3ff696e4732750b7bdc58461 Mon Sep 17 00:00:00 2001
From: Matthew Meola <mlmeola@gmail.com>
Date: Sat, 17 Mar 2018 15:26:23 -0400
Subject: [PATCH] Use py bytes to interact with ctypes

Changes string args to ctypes fns to bytes and then encodes output as
utf-8.
---
 trashcli/list_mount_points.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/trashcli/list_mount_points.py b/trashcli/list_mount_points.py
index f0f9cb8..5f66b75 100644
--- a/trashcli/list_mount_points.py
+++ b/trashcli/list_mount_points.py
@@ -2,7 +2,7 @@
 
 def mount_points():
     try:
-        return list(mount_points_from_getmnt())
+        return list(s.decode('utf-8') for s in mount_points_from_getmnt())
     except AttributeError:
         return mount_points_from_df()
 
@@ -60,9 +60,9 @@ class mntent_struct(Structure):
     libc.fopen.restype = c_void_p
     libc.fclose.argtypes = [c_void_p]
 
-    f = libc.fopen("/proc/mounts", "r")
+    f = libc.fopen(b"/proc/mounts", b"r")
     if f==None:
-        f = libc.fopen("/etc/mtab", "r")
+        f = libc.fopen(b"/etc/mtab", b"r")
         if f == None:
             raise IOError("Unable to open /proc/mounts nor /etc/mtab")
 
